### PR TITLE
chore_: remove CLI option  MEDIA_HTTPS

### PR DESCRIPTION
**Pull request by qfrank**
relate [comment](https://github.com/status-im/status-go/pull/5943#issuecomment-2485160525)

relate mobile [PR](https://github.com/status-im/status-mobile/pull/21648)

This is a followup PR to [PR](https://github.com/status-im/status-go/pull/6060) that address feedback from @igor-sirotin 

Major changes:
- remove global var `useHTTP`
- remove CLI option  MEDIA_HTTPS
- minor refactors

### CODE REVIEW

File: node/get_status_node.go [hunk 1]
```diff
@@ -269,6 +271,10 @@ func (n *StatusNode) StartWithOptions(config *params.NodeConfig, options StartOp
 	return nil
 }
 
+func (n *StatusNode) UpdateMediaServerEnableTLS(enableTLS bool) {
```
Comment (igor-sirotin) [inline node/get_status_node.go]:
```suggestion
func (n *StatusNode) SetMediaServerEnableTLS(enableTLS bool) {
```

File: mobile/status.go [hunk 2]
```diff
@@ -84,9 +84,13 @@ func initializeApplication(requestJSON string) string {
 	providers.MixpanelAppID = request.MixpanelAppID
 	providers.MixpanelToken = request.MixpanelToken
 
-	datadir := request.DataDir
+	mediaServerEnableTLS := true
+	if request.MediaServerEnableTLS != nil {
+		mediaServerEnableTLS = *request.MediaServerEnableTLS
+	}
+	statusBackend.StatusNode().UpdateMediaServerEnableTLS(mediaServerEnableTLS)
```
Comment (igor-sirotin) [inline mobile/status.go]: I guess it's `true` by default anyway?
```suggestion
	if request.MediaServerEnableTLS != nil {
		statusBackend.StatusNode().UpdateMediaServerEnableTLS(*request.MediaServerEnableTLS)
	}
```

File: server/server_media.go [hunk 3]
```diff
@@ -14,32 +14,38 @@ import (
...
 // NewMediaServer returns a *MediaServer
-func NewMediaServer(db *sql.DB, downloader *ipfs.Downloader, multiaccountsDB *multiaccounts.Database, walletDB *sql.DB) (*MediaServer, error) {
-	var cert *tls.Certificate
-	if !UseHTTP {
-		err := generateMediaTLSCert()
-		if err != nil {
-			return nil, err
-		}
-		cert = globalMediaCertificate
+func NewMediaServer(db *sql.DB, downloader *ipfs.Downloader, multiaccountsDB *multiaccounts.Database, walletDB *sql.DB, useHTTP bool) (*MediaServer, error) {
```
Comment (igor-sirotin) [inline server/server_media.go]: It's too many arguments here already. Can you please make it `[]Options`?
Also, if you make it `opts ...Option`, you'd not need to change all usages of `NewMediaServer`, only the ones with non-default behaviour.

File: server/server_media.go [hunk 4]
```diff
@@ -14,32 +14,38 @@ import (
...
 	db              *sql.DB
 	downloader      *ipfs.Downloader
 	multiaccountsDB *multiaccounts.Database
 	walletDB        *sql.DB
+
+	// UseHTTP controls whether the media server uses HTTP instead of HTTPS.
+	// Set to true to avoid TLS certificate issues with react-native-fast-image
+	// on Android, which has limitations with dynamic certificate updates.
+	// Pls check doc/use-status-backend-server.md in status-mobile for more details
+	useHTTP bool
```
Comment (igor-sirotin) [inline server/server_media.go]: I still think `enableTLS` is a better name 🙂

**Review by igor-sirotin — APPROVED** [review]

**Review by ilmotta — APPROVED** [review]

**Comment by qfrank** [comment]
All feedback addressed @igor-sirotin Thank you all for reviewing cc @ilmotta

**Review by igor-sirotin — APPROVED** [review]

**Pull request merged**